compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -878,6 +878,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             format!("impl {}", all_bounds_str),
             Applicability::MaybeIncorrect,
         );
+        err.note("the caller chooses the value of a type parameter");
```

Comment:
Maybe something like..

> the type of `T` is chosen by the caller

..would be clearer?

I'm not keen on describing type parameters as having a "value", I think that could lead to confusion.